swo_mpt_api/billing/journal_client.py

```diff
@@ -17,10 +18,31 @@ def query(self, rql=None) -> HttpQuery[JournalAttachment]:
         url = f"/billing/journals/{self.journal_id}/attachments"
         return HttpQuery(self._client, url, rql)
 
-    def upload(self, attachment: IO) -> JournalAttachment:
-        response = self._client.post(
-            f"/billing/journals/{self.journal_id}/attachments", files={"file": attachment}
-        )
+    def upload(
+        self, file, mimetype:str, filename:str|None=None, attachment:JournalAttachment|None=None
```

Comment:
please follow styleguide convention here, not sure why ruff didn't fix it for you, but there should be whitespaces between `=` `|` `:` symbols